iocage/lib/Config/Jail/Properties/ResourceLimit.py

```diff
@@ -0,0 +1,192 @@
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Jail config resource limit."""
+import typing
+import iocage.lib.errors
+
+properties: typing.List[str] = [
+    "cputime",
+    "datasize",
+    "stacksize",
+    "coredumpsize",
+    "memoryuse",
+    "memorylocked",
+    "maxproc",
+    "openfiles",
+    "vmemoryuse",
+    "pseudoterminals",
+    "swapuse",
+    "nthr",
+    "msgqqueued",
+    "msgqsize",
+    "nmsgq",
+    "nsem",
+    "nsemop",
+    "nshm",
+    "shmsize",
+    "wallclock",
+    "pcpu",
+    "readbps",
+    "writebps",
+    "readiops",
+    "writeiops"
+]
+
+
+class ResourceLimitValue:
+    """Model of a resource limits value."""
+
+    amount: typing.Optional[str]
+    action: typing.Optional[str]
+    per: typing.Optional[str]
+
+    def __init__(self) -> None:
+        self.amount = None
+        self.action = None
+        self.per = None
+
+    def _parse_resource_limit(
+        self,
+        value: str
+    ) -> typing.Tuple[str, str, str]:
+
+        if ("=" not in value) and (":" not in value):
+            # simplified syntax vmemoryuse=128M
+            amount = value
+            action = "deny"
+            per = "jail"
+        elif "=" in value:
+            # rctl syntax
+            action, _rest = value.split("=", maxsplit=1)
+            amount, per = _rest.split("/", maxsplit=1)
+        elif ":" in value:
+            # iocage legacy syntax
+            amount, action = value.split(":", maxsplit=1)
+            per = "jail"
+        else:
+            raise ValueError("invalid syntax")
+
+        if (amount == "") or (action == "") or (per == ""):
+            raise ValueError("value may not be empty")
+
+        return amount, action, per
+
+    def __str__(self) -> str:
+        """
+        Return the resource limit value in string format.
+
+        When self.per is "jail" the legacy compatible format is used.
+        """
+        if self.per == "jail":
+            return f"{self.amount}:{self.action}"
+        else:
+            return self.limit_string
+
+    @property
+    def limit_string(self) -> str:
+        """Return the limit string in rctl syntax."""
+        return f"{self.action}={self.amount}/{self.per}"
+
+
+_ResourceLimitInputType = typing.Optional[
+    typing.Union[str, ResourceLimitValue]
+]
+
+
+class ResourceLimitProp(ResourceLimitValue):
+    """Special jail config property for resource limits."""
+
+    def __init__(
+        self,
+        config: typing.Optional[
+            'iocage.lib.Config.Jail.BaseConfig.BaseConfig'
+        ]=None,
+        property_name: str="ip4_address",
```

Comment:
why is this the default `property_name`?

Reply:
There's no need for a default. Will remove.